doc/Data Plane Code Generation/ServiceDrivenEvolution.md

```diff
@@ -0,0 +1,177 @@
+
+## A new body type is added
+
+Generated code in a V1 client with only `application/json` as the content type:
+
+``` C#
+public virtual async Task<Response> PostParametersAsync(RequestContent content, RequestContext context = null)
+{
+    Argument.AssertNotNull(content, nameof(content));
+
+    using var scope = ClientDiagnostics.CreateScope("ParamsClient.PostParameters");
+    scope.Start();
+    try
+    {
+        using HttpMessage message = CreatePostParametersRequest(content, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    catch (Exception e)
+    {
+        scope.Failed(e);
+        throw;
+    }
+}
+
+internal HttpMessage CreatePostParametersRequest(RequestContent content, RequestContext context)
+{
+    var message = _pipeline.CreateMessage(context, ResponseClassifier200);
+    var request = message.Request;
+    request.Method = RequestMethod.Post;
+    var uri = new RawRequestUriBuilder();
+    uri.Reset(_endpoint);
+    uri.AppendPath("/serviceDriven/parameters", false);
+    request.Uri = uri;
+    request.Headers.Add("Accept", "application/json");
+    request.Headers.Add("Content-Type", "application/json");
+    request.Content = content;
+    return message;
+}
+```
+
+Generated code in a V2 client with `image/jpeg` as a new content type in addition to `application/json`:
+
+``` C#
+...
+/// <param name="contentType"> Body Parameter content-type. Allowed values: &quot;application/json&quot; | &quot;image/jpeg&quot;. </param>
+...
+public virtual async Task<Response> PostParametersAsync(RequestContent content, ContentType contentType, RequestContext context = null)
+{
+    Argument.AssertNotNull(content, nameof(content));
+
+    using var scope = ClientDiagnostics.CreateScope("ParamsClient.PostParameters");
+    scope.Start();
+    try
+    {
+        using HttpMessage message = CreatePostParametersRequest(content, contentType, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    catch (Exception e)
+    {
+        scope.Failed(e);
+        throw;
+    }
+}
+
+internal HttpMessage CreatePostParametersRequest(RequestContent content, ContentType contentType, RequestContext context)
+{
+    var message = _pipeline.CreateMessage(context, ResponseClassifier200);
+    var request = message.Request;
+    request.Method = RequestMethod.Post;
+    var uri = new RawRequestUriBuilder();
+    uri.Reset(_endpoint);
+    uri.AppendPath("/serviceDriven/parameters", false);
+    request.Uri = uri;
+    request.Headers.Add("Accept", "application/json");
+    request.Headers.Add("Content-Type", contentType.ToString());
+    request.Content = content;
+    return message;
+}
+```
+
+Since the V2 client added a required parameter, it breaks the code that is written for V1. We can manually add an overload method that has the same signature as V1 to make it backward compatible:
```

Comment:
```suggestion
Since the V2 client added a required parameter to an existing client method, it breaks the code that is written for the V1 client. We can manually add an overload method that has the same signature as V1 to make it backward compatible:
```